src/main/java/com/minecolonies/coremod/entity/ai/citizen/beekeeper/EntityAIWorkBeekeeper.java

```diff
@@ -0,0 +1,460 @@
+          new AITarget(BEEKEEPER_HARVEST, this::harvestHoney, TICKS_SECOND)
+        );
+    }
+
+    public boolean maxAnimals(final List<BeeEntity> allAnimals)
+    {
+        if (getOwnBuilding() != null)
+        {
+            final int numOfBeesInHive = getOwnBuilding(BuildingBeekeeper.class)
+                                          .getHives()
+                                          .stream()
+                                          .map(world::getTileEntity)
+                                          .filter(Objects::nonNull)
+                                          .map(BeehiveTileEntity.class::cast)
+                                          .mapToInt(BeehiveTileEntity::func_226971_j_)
+                                          .sum();
+            final int numOfAnimals = allAnimals.size();
+            final int maxAnimals = getOwnBuilding().getBuildingLevel() * BEES_PER_LEVEL;
+
+            return (numOfAnimals + numOfBeesInHive) > maxAnimals;
+        }
+        return true;
+    }
+
+    /**
+     * Prepares the beekeeper for herding
+     *
+     * @return The next {@link IAIState}.
+     */
+    private IAIState prepareForHerding()
+    {
+        setDelay(DECIDING_DELAY);
+        if (getOwnBuilding(BuildingBeekeeper.class).shouldHarvestHoneycombs())
+        {
+            if (checkForToolOrWeapon(ToolType.SHEARS))
+            {
+                return getState();
+            }
+        }
+
+        checkIfRequestForItemExistOrCreateAsynch(new ItemStack(Items.GLASS_BOTTLE));
+        checkIfRequestForTagExistOrCreateAsynch(ItemTags.field_226159_I_, 16, FLOWERS);
+
+        return DECIDE;
+    }
+
+    /**
+     * Redirects the beekeeper to their building.
+     *
+     * @return The next {@link IAIState}.
+     */
+    private IAIState startWorkingAtOwnBuilding()
+    {
+        worker.getCitizenStatusHandler().setLatestStatus(new TranslationTextComponent(TranslationConstants.COM_MINECOLONIES_COREMOD_STATUS_WORKER_GOINGTOHUT));
+        if (walkToBuilding())
+        {
+            return getState();
+        }
+        return PREPARING;
+    }
+
+    /**
+     * Decides what job the beekeeper should switch to, breeding or harvesting.
+     *
+     * @return The next {@link IAIState} the beekeeper should switch to, after executing this method.
+     */
+    private IAIState decideWhatToDo()
+    {
+        setDelay(DECIDING_DELAY);
+
+        final Optional<BlockPos> hive = getOwnBuilding(BuildingBeekeeper.class)
+                                          .getHives()
+                                          .stream()
+                                          .filter(pos -> BeehiveTileEntity.func_226964_a_(world.getBlockState(pos)) >= 5)
+                                          .findFirst();
+        if (hive.isPresent())
+        {
+            return BEEKEEPER_HARVEST;
+        }
+
+        final List<BeeEntity> bees = new ArrayList<>(searchForAnimals());
+
+        if (bees.isEmpty())
```

Comment:
Should probably set an interaction here, if they can't find bees within a time they gotta tell the player about this.

Reply:
Better probably if hive is empty. Then add the interaction.